operate/cli.py

```diff
@@ -663,6 +666,89 @@ async def _stop_service_onchain(request: Request) -> JSONResponse:
             )
         )
 
+    @app.post("/api/services/{service}/onchain/withdraw")
+    @with_retries
+    async def _withdraw_onchain(request: Request) -> JSONResponse:
+        """Withdraw all the funds from a service."""
+        if not operate.service_manager().exists(service=request.path_params["service"]):
+            return service_not_found_error(service=request.path_params["service"])
+        if operate.password is None:
+            return USER_NOT_LOGGED_IN_ERROR
+        
+        withdrawal_address = (await request.json()).get("withdrawal_address")
+        if withdrawal_address is None:
+            return JSONResponse(
+                content={"error": "withdrawal_address is required"},
+                status_code=400,
+            )
+
+        try:
+            service_manager = operate.service_manager()
+            service = service_manager.load_or_create(hash=request.path_params["service"])
+
+            service_manager.terminate_service_on_chain_from_safe(
+                hash=request.path_params["service"],
+                chain_id=service.home_chain_id,
+                is_withdrawing=True,
+            )
+
+            service_manager.drain_service(
+                hash=request.path_params["service"],
+                withdrawal_address=withdrawal_address
+            )
+
+            # drain OLAS from the master safe
+            chain_config = service.chain_configs[service.home_chain_id]
+            master_wallet: EthereumMasterWallet = service_manager.wallet_manager.load(
+                ledger_type=chain_config.ledger_config.type
+            )
+            ocm = service_manager.get_on_chain_manager(ledger_config=chain_config.ledger_config)
```

Comment:
On-chain manager is used for EVM operations without Safe (legacy code). This should not be used, in favour of EthSafeTxBuilder

Reply:
All OnChainManager code will be removed, as we are not considering use cases without Safe. cc @oaksprout

Reply:
I agree with @jmoreira-valory, lets use `EthSafeTxBuilder`.